jgiven-core/src/test/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportModelVisitorTest.java

```diff
@@ -0,0 +1,163 @@
+package com.tngtech.jgiven.report.asciidoc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.tngtech.jgiven.report.ReportBlockConverter;
+import com.tngtech.jgiven.report.model.CasesTable;
+import com.tngtech.jgiven.report.model.ExecutionStatus;
+import com.tngtech.jgiven.report.model.ReportModel;
+import com.tngtech.jgiven.report.model.ReportStatistics;
+import com.tngtech.jgiven.report.model.ScenarioCaseModel;
+import com.tngtech.jgiven.report.model.ScenarioModel;
+import com.tngtech.jgiven.report.model.StepModel;
+import com.tngtech.jgiven.report.model.StepStatus;
+import com.tngtech.jgiven.report.model.Word;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AsciiDocReportModelVisitorTest {
+    private final ReportBlockConverter blockConverter = new MyFakeReportBlockConverter();
+
+    private AsciiDocReportModelVisitor reportModelVisitor;
+
+    @Before
+    public void setUp() {
+        ReportStatistics reportStatistics = new ReportStatistics();
+        reportModelVisitor = new AsciiDocReportModelVisitor(blockConverter, reportStatistics);
+    }
+
+    @Test
+    public void visitTheReportModel() {
+        // given
+        StepModel stepModel = new StepModel();
+        stepModel.addIntroWord(Word.introWord("Given"));
+        stepModel.addWords(new Word("some"), new Word("state"));
+
+        ScenarioCaseModel scenarioCaseOne = new ScenarioCaseModel();
+        scenarioCaseOne.addStep(stepModel);
+
+        ScenarioModel currentScenarioModel = new ScenarioModel();
+        currentScenarioModel.addCase(scenarioCaseOne);
+        currentScenarioModel.addCase(new ScenarioCaseModel());
+
+        ReportModel reportModel = new ReportModel();
+        reportModel.addScenarioModel(currentScenarioModel);
+
+        // when
+        reportModel.accept(reportModelVisitor);
+
+        // then
+        assertThat(reportModelVisitor.getResult())
+                .isEqualTo(ImmutableList.of(
+                        "convertFeatureHeaderBlock",
+                        "convertScenarioHeaderBlock",
+                        "convertCaseHeaderBlock",
+                        "convertFirstStepBlock",
+                        "convertCaseHeaderBlock",
+                        "convertScenarioFooterBlock"));
+    }
+
+    @Test
+    public void visitTheReportModelWithTwoSections() {
```

Comment:
shouldn't this be spelt `visitAScenarioWithTwoScenarioCases`? Speaking of which, checking that two Scenarios in a report are handled gracefully might also be a good idea

Reply:
Yeah the test did check multiple things. Now there are more tests that hopefully only check for exactly one situation.